execute/README.md

```diff
@@ -0,0 +1,458 @@
+
+A narrow transformation that is stateless is implemented using the `NarrowTransformation` interface.
+
+    type NarrowTransformation interface {
+        Process(chunk table.Chunk, d *TransportDataset, mem memory.Allocator) error
+    }
+
+The `Process` method is implemented to take a table chunk, transform it using the memory allocator for new allocations, and then send it to be processed by the dataset.
+A skeleton implementation is shown below:
+
+    func (t *MyNarrowTransformation) Process(chunk table.Chunk, d *TransportDataset, mem memory.Allocator) error {
+        out, err := t.process(chunk, mem)
+        if err != nil {
+            return err
+        }
+        return d.Process(out)
+    }
+
+    func (t *MyNarrowTransformation) process(chunk table.Chunk, mem memory.Allocator) (table.Chunk, error) {
+        /* transformation-specific logic */
+    }
+
+Some examples where this version of the narrow transformation is used: [filter()](https://github.com/influxdata/flux/blob/master/stdlib/universe/filter.go) and [fill()](https://github.com/influxdata/flux/blob/master/stdlib/universe/fill.go).
+
+An alternative is used when we need to maintain state between chunks.
+
+    type NarrowStateTransformation interface {
+        Process(chunk table.Chunk, state interface{}, d *TransportDataset, mem memory.Allocator) (interface{}, bool, error)
+    }
+
+The first time the group key is encountered, the state will be nil.
+It should be initialized by the `Process` method.
+The new state will be stored if it is returned along with the second return argument being true.
+If the second return argument is false, the state will not be modified.
+It is both ok and expected that the `interface{}` will be a pointer to a struct and will be modified in a mutable way.
+The state is not expected to be immutable.
+
+A skeleton implementation is shown below:
+
+    type myState struct { ... }
+
+    func (t *MyNarrowTransformation) Process(chunk table.Chunk, state interface{}, d *TransportDataset, mem memory.Allocator) error {
+        state := t.loadState(state)
+        out, err := t.process(chunk, state, mem)
+        if err != nil {
+            return nil, false, err
+        }
+
+        if err := d.Process(out); err != nil {
+            return nil, false, err
+        }
+        return state, true, nil
+    }
+
+    func (t *MyNarrowTransformation) loadState(state interface{}) *myState {
+        if state == nil {
+            return &myState{}
+        }
+        return state.(*myState)
+    }
+
+    func (t *MyNarrowTransformation) process(chunk table.Chunk, state *myState, mem memory.Allocator) (table.Chunk, error) {
+        /* transformation-specific logic */
+    }
+
+Some examples where this version of the narrow transformation is used: [derivative()](https://github.com/influxdata/flux/blob/master/stdlib/universe/derivative.go).
+
+### Group Transformation
+
+TODO: Write this section.
```

Comment:
Are there issues open for this (and the following) `TODO`? There's little more awkward than seeing a doc years old that has `TODO: document this part`. *glares at many of those online rust books*

Reply:
Not at the moment. I figured I would just circle back and get these once the related issues had been done.